graph-apps/graph-apps.adoc

```diff
@@ -0,0 +1,675 @@
+
+Whichever framework you choose to build your application, you will interact with the Graph the same way; by running a Cypher Query through the https://github.com/neo4j/neo4j-javascript-driver[Official JavaScript drivers].
+There is plenty of content over on the https://neo4j.com/developer/javascript/[JavaScript Developer Pages] so we will not cover this in too much detail, only the nuances around how the driver works with the Neo4j Desktop API.
+
+
+== Enabling Developer Tools
+
+Neo4j Desktop comes with a Development App that can be used while developing and debugging your Graph App.
+To enable Development Mode, click on the Settings image:{img}settings.png[Settings Icon, title="Settings Icon"] icon in the bottom left hand corner of the screen to reveal the Settings pane.
+
+At the bottom of the Settings pane, you will see a *Developer Tools* form.
+Click *Enable Development Mode* and add fill out the Entry Point and Root Path.
+
+image:{img}developer-tools.png[Developer Tools window with Enable development mode checkbox and fields for Entry Point and Root Path, title="Developer Tools form"]
+
+- *Entry Point*: This is the path to your app, either a location of a HTML file (for example `file:///Users/me/my-graph-app/index.html`) or a HTTP url where your app is being served (for example `http://localhost:8080`).
+- *Root Path*: This is the path to the root of your development project.  If you are using #background-processes[Background Processes], this should be the location of the files that contain your Java or Node.js processes.  - if in doubt, you can just set it to `/`.
+
+== Getting the Context
+
+If you require access to the the list of projects and databases that the user has configured in Neo4j Desktop, you can access this via the Context API.
+There are two ways of accessing Neo4j Desktop's context, either through the Injected API or using GraphQL.
+
+Neo4j Desktop provides information about the current desktop environment via the `neo4jDesktopApi` object that is injected into the window.  This will give you [TODO: configuration] of Neo4j Desktop, including all Projects and Databases.
+
+
+==== Injected API
+
+The _Injected_ API is provided through the `neo4jDesktopApi` object.  The `getContext()` method returns a Promise which in turn resolves to an object containing a `projects` key.  Each Project contains a list of Neo4j Databases and a list of files.
+
+```js
+if ( window.neo4jDesktopApi ) {
+  neo4jDesktopApi.getContext()
+    .then(context => {
+      // Access to the list of projects will be through context.projects
+    })
+}
+```
+
+If the context should change for any reason, for example if the active database changes or has been stopped, an event will be fired.  You can listen for these changes using the `onContextUpdate` method on the API.  This will provide you an Event object, the both the new and old contexts.
```

Comment:
> This will provide you an Event object, the both the new and old contexts.

should be

>This will provide you an Event object that contains both the new and old contexts.

I think?

Reply:
The callback function passes the event, the new context and the old context.  Maybe this wasn't clear enough?

```(event, newContext, oldContext) => {```